src/main/java/com/drajer/bsa/ehr/service/impl/BackendAuthorizationServiceImpl.java

```diff
@@ -0,0 +1,189 @@
+    return new JSONObject(Objects.requireNonNull(response.getBody()));
+  }
+
+  public String getTokenEndpoint(String url) {
+    RestTemplate resTemplate = new RestTemplate();
+    try {
+      ResponseEntity<String> res =
+          resTemplate.getForEntity(String.format("%s/%s", url, WELL_KNOWN), String.class);
+      JSONArray result = JsonPath.read(res.getBody(), "$.token_endpoint");
+      return result.get(0).toString();
+    } catch (Exception e1) {
+      try {
+        ResponseEntity<String> res =
+            resTemplate.getForEntity(String.format("%s/metadata", url), String.class);
+        // jsonpath allows filtering through lists with '?', where '@' represents the current
+        // element
+        JSONArray result =
+            JsonPath.read(
+                res.getBody(),
+                "$.rest[?(@.mode == 'server')].security"
+                    + ".extension[?(@.url == '"
+                    + OAUTH_URIS
+                    + "')]"
+                    + ".extension[?(@.url == 'token')].valueUri");
+        return result.get(0).toString();
+      } catch (Exception e2) {
+        logger.error("Error in Getting the TokenEndpoint for the client: {}", url, e2);
+        throw e2;
+      }
+    }
+  }
+
+  public String getRegistrationEndpoint(String url) {
+    RestTemplate resTemplate = new RestTemplate();
+    try {
+      ResponseEntity<String> res =
+          resTemplate.getForEntity(String.format("%s/%s", url, WELL_KNOWN), String.class);
+      JSONArray result = JsonPath.read(res.getBody(), "$.registration_endpoint");
+      return result.get(0).toString();
+    } catch (Exception ex1) {
+      try {
+        ResponseEntity<String> res =
+            resTemplate.getForEntity(String.format("%s/metadata", url), String.class);
+        // jsonpath allows filtering through lists with '?', where '@' represents the current
+        // element
+        JSONArray result =
+            JsonPath.read(
+                res.getBody(),
+                "$.rest[?(@.mode == 'server')].security"
+                    + ".extension[?(@.url =='"
+                    + OAUTH_URIS
+                    + "')]"
+                    + ".extension[?(@.url == 'register')].valueUri");
+        return result.get(0).toString();
+      } catch (Exception ex2) {
+        logger.error("Error trying to access ehr metadata endpoint");
+        throw ex2;
+      }
+    }
+  }
+
+  public String generateJwt(String clientId, String aud) throws KeyStoreException {
+
+    KeyStore ks = KeyStore.getInstance(KeyStore.getDefaultType());
+    char[] passwordChar = password.toCharArray();
+    try {
+      InputStream store = new FileInputStream(jwksLocation);
+      ks.load(store, passwordChar);
+      store.close();
+      Key key = ks.getKey(alias, passwordChar);
+
+      return Jwts.builder()
+          .setIssuer(clientId)
+          .setSubject(clientId)
+          .setAudience(aud)
+          .setExpiration(
+              new Date(
```

Comment:
This is really silly but the build actually fails when there are style violations. There's apparently one here and one extra blank line somewhere. Rather than worry about them individually you should be able to resolve them by running `mvn fmt:format`